src/org/thoughtcrime/securesms/database/EmojiReactionDatabase.java

```
@@ -0,0 +1,123 @@
+        SQLiteDatabase db = databaseHelper.getWritableDatabase();
+        db.beginTransaction();
+        try {
+            String hashedId = messageRecord.getHashedId();
+
+            ContentValues contentValues = new ContentValues();
+            contentValues.put(REACTION, reaction);
+            contentValues.put(HASHED_ID, hashedId);
+
+            String where = HASHED_ID + "= '" + hashedId + "'";
+
+            // Update the reaction in the DB for the message with the specified hashed id.
+            int update = db.update(TABLE_NAME, contentValues, where, null);
+
+            // If update == 0, reaction does not exists for the hashed id, and is inserted.
+            if(update == 0){
+                db.insert(TABLE_NAME,null, contentValues);
+            }
+
+            db.setTransactionSuccessful();
+        }catch(Exception e){
+            Log.d(TAG, e.getMessage());
+        }finally{
+            db.endTransaction();
+        }
+
+        // Notify the conversation listener of a change, to refresh the conversation.
+        long threadId = messageRecord.getThreadId();
+        notifyConversationListeners(threadId);
+    }
+
+    //Add emoji reaction to the database (method used by RECEIVER)
+    public void setMessageReaction(String hashedId, String reaction, long threadId){
+        SQLiteDatabase db = databaseHelper.getWritableDatabase();
+        db.beginTransaction();
+
+        try{
+            ContentValues contentValues = new ContentValues();
+            contentValues.put(REACTION, reaction);
+            contentValues.put(HASHED_ID, hashedId);
+
+            String where = HASHED_ID + "= '" + hashedId + "'";
+            int update = db.update(TABLE_NAME, contentValues, where, null);
+
+            if(update == 0){
+                db.insert(TABLE_NAME,null, contentValues);
+            }
+
+            db.setTransactionSuccessful();
+        }catch(Exception e){
```

I would change catching exception e to catching the specific exceptions instead, to make it more clear what kind of exception these operations can throw/and to not squash other unexpected exceptions.